### PR TITLE
add pre-commit github action

### PR DESCRIPTION
**Pull request by grahamgower**

### CODE REVIEW

**Comment by grahamgower** [comment]
I think this is ready for merging. @mufernando are you sure you're ok with rebasing your PR onto this monstrosity?

File: update_ensembl_data.py [hunk 1]
```diff
@@ -40,8 +40,9 @@ def write_genome_data(self, ensembl_id):
         if len(id) != 6:
             raise ValueError(f"Cannot extract six character id from {ensembl_id}")
         if not os.path.exists(dir):
-            raise ValueError(f"Directory {id} corresponding to {ensembl_id} does" +
-            "not exist")
+            raise ValueError(
```
Comment (jeromekelleher) [inline update_ensembl_data.py:43]: We can now remove the `` subprocess.check_call(["black", "-q", path])`` above according to the TODO. Might as well tack a commit on here to do that as well.

**Review by jeromekelleher — APPROVED** [review]
HUGE improvement, thanks @grahamgower! I've skimmed the changes and all looks good. One comment below.

File: update_ensembl_data.py (hunk 1 above)
Comment (grahamgower) [reply update_ensembl_data.py:38]: Ah, yep, I've chucked that in too.

**Comment by mufernando** [comment]
LGTM!

**Pull request merged**